app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -107,33 +98,32 @@ class LoginActivity: AppCompatActivity() {
     private fun handleFacebookAccessToken(accessToken: AccessToken?) {
         val credential = FacebookAuthProvider.getCredential(accessToken!!.token)
         mAuth.signInWithCredential(credential)
-                .addOnCompleteListener(this) { task: Task<AuthResult?> ->
-                    if (task.isSuccessful) {
-                        val user = mAuth.currentUser
-                        saveUser(user)
-                    } else {
-                        Toast.makeText(this@LoginActivity, "Authentication failed." + task.exception,
-                                Toast.LENGTH_SHORT).show()
-                    }
+                .addOnCompleteListener(this) { task ->
+                    if (task.isSuccessful)
+                        saveUser(mAuth.currentUser)
+                    else
+                        toast("Authentication failed." + task.exception)
                 }
     }
 
+
     private fun saveUser(user: FirebaseUser?) {
         val model = UserModel()
         model.name = user!!.displayName
         model.userId = user.uid
         model.rejectedPost = false
         model.email = user.email
-        FirebaseFirestore.getInstance().collection("user").document(user.uid).set(model)
+        FirebaseFirestore.getInstance().collection(user1).document(user.uid).set(model)
         getSharedPreferences("MAIN", Context.MODE_PRIVATE).edit().putString("uid", user.uid).apply()
         createNotificationChannel()
-        val editor: SharedPreferences.Editor = getSharedPreferences("GENERAL", Context.MODE_PRIVATE).edit()
+        val editor = getSharedPreferences("GENERAL", Context.MODE_PRIVATE).edit()
```

Comment:
make "GENERAL" a constant in constants.kt and give sutale name like SP_XYZ. Use that for all shared prefrences constants.